exporter/tanzuobservabilityexporter/metrics.go

```diff
@@ -345,3 +350,195 @@ func (s *sumConsumer) pushNumberDataPoint(
 		*errs = append(*errs, err)
 	}
 }
+
+// histogramReporting takes care of logging and internal metrics for histograms
+type histogramReporting struct {
+	logger                   *zap.Logger
+	malformedHistograms      counter
+	leInUse                  counter
+	noAggregationTemporality counter
+}
+
+// newHistogramReporting returns a new histogramReporting instance.
+// log messages are sent to logger.
+func newHistogramReporting(logger *zap.Logger) *histogramReporting {
```

Comment:
Looking at the code, it seems like I need to implement the MeterImpl interface.  As you know MeterImpl has 3 methods: RecordBatch, NewSyncInstrument, NewAsyncInstrument.  It looks like RecordBatch might do what I need which is to send a bunch of metrics to tanzu observability. But, I am not sure how to implement NewSyncInstrument and NewAsyncInstrument. I don't understand the contract of those two methods.  With that said, I still need a way to count events, so I was looking for something with an `Increment()` or `Inc()` method, but `Measurement` doesn't seem to have this.  In short, I haven't yet figured out how to reuse this code.

Reply:
Sorry, my comment was really about the logger parameter. Instead of receiving a logger, you'd receive a `component.TelemetrySettings`, which does contain a logger you can use.

About the reporting types (`histogramReporting`), take a look at my other comment: it points to another component that creates data points for operations performed within the component itself. As Bogdan said, it might be strange to make telemetry data available via OpenCensus/OpenTelemetry when you are at a vendor exporter, but consider that your users might want to monitor the collector in a different way than their workloads.

Reply:
So if I understand correctly, I should use a component.TelemetrySettings instead of a zap.Logger whenever I plan to log events and update internal metrics.  Question:  zap.NewNop() is great for creating a no-op logger when you don't have a *real* logger such as in tests.  Is there a preferred way to create a No-op component.TelemetrySettings?  I could just say `component.TelemetrySettings{Logger: zap.NewNop()}` but if I do that then I am not initializing the TracerProvider and MeterProvider fields.

Reply:
Yes: `componenttest.NewNopTelemetrySettings()` from `go.opentelemetry.io/collector/component/componenttest`

https://github.com/open-telemetry/opentelemetry-collector/blob/main/component/componenttest/nop_telemetry.go#L26

Reply:
Done.